compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -650,6 +650,10 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
         if tcx.lazy_normalization() {
             constant
         } else {
+            if !needs_normalization(&constant, self.param_env.reveal()) {
```

Comment:
This could be merged into the above condition

Reply:
of course